### ISSUE TITLE
mobile interfaces

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#1114]
it's clear that the current interfaces are difficult to use on mobile devices. rather than try to make everything work on mobiles, what are the most common or necessary things that we need to do on mobiles, & can we make mobile interfaces for those things?

- sending a stream: it's very important that players are able to easily send a stream from their mobile phone or tablet to a stage. previously we could do this easily using Larix or similar app, with the QR code set-up. the jitsi streaming sort of works but is difficult as the stream needs to be placed on the stage using the device that is sending the stream, & this is very cumbersome on a small screen. ideally, a stream would be sent from a smartphone and another player could place that stream on the stage.
- uploading media: if you have taken a photo, video or audio recording with your phone, it's convenient to be able to upload it directly from the phone to UpStage. this is possible via the browser interface, but it's not easy because of the size of the screen. ideally there should be a mobile upload interface that enables a simple upload process.
- admin tasks: sometimes there can be an urgent situation where a player account needs to be approved or something needs to be checked or fixed by an admin, when they may be travelling or away from their computer. this can be done via the browser interface on a phone, but it's cumbersome.

@vickis - anything else you can think of?
@elisa - any ideas about mobile layout for these things?

**Comment by elisadcg** [comment]
Ok I look how Instagram, Snapchat and Tiktok do.
First we have to give access the possibility from a smartphone to connect directly to the studio/media > send a direct stream 
![send-stream-mobile01](https://github.com/upstage-org/upstage/assets/7611913/0026161b-89ad-4ae6-b2fe-755431df5d03)
The user just it the "+" to able his camera, select which camera and hit "send stream" (or cancel). Then 
![send-stream-mobile02](https://github.com/upstage-org/upstage/assets/7611913/5851218d-2bc0-41b0-9159-0aa012dd2dfe)
the user can send the stream and manage (mayve) the flash and the audio of his stream.
We can put this direct icon at the bottom, or at the top : 
 ![send-stream-mobile02b](https://github.com/upstage-org/upstage/assets/7611913/a6885f36-f0fe-4a71-8d7c-c1c48fea8a67)
And when the user want stop the live, he hit the switch button stop stream and go back to the previous screens. 
![send-stream-mobile03](https://github.com/upstage-org/upstage/assets/7611913/a787fc43-2208-446d-bf01-d6e33e01ecee)

**Comment by helenvarley** [comment]
the idea is that the person with the phone would not need to also be on the stage, another player would need to put their stream on (this was possible with rtmp streams, currently is not possible with individual streams, so something for the developers to find a solution for).

**Comment by elisadcg** [comment]
That's why the live stream go to media.
The media have to be already placed on a stage. And before the stream. So
that's need an image representing the stream waiting for the stream.

But we probably need more technical information from developers to know
what they need to put a live stream on a stage. Then we can imagine how it
could work as a player.

**Comment by vickis** [comment]
Yes the interface you show would be perfect for rmtp and as we know worked
from apps like larix
The requirement will be the ability to connect to jitsi bridge as an
‘object’ that can be placed by other players

This might be some function of a stage that needs thought as usually if I
placed a jitsi stream on stage then it would request access to my camera
and mic :)

**Comment by helenvarley** [comment]
right - needs some kind of container that another player can place on stage as a media item, that will then display the stream sent from the phone (as it worked with rtmp). so a fun new challenge for the devs to see if it can work in another way - jitsi bridge? webrtc?

**Comment by vickis** [comment]
would be great if potential to have the 'container' and choose the content
source eg. webrtc, jitsi or larix or....

**Comment by gloriajw** [comment]
This sounds like it's dipping into the really challenging research project work. I don't want devs here to get discouraged by this challenge.  This can be overwhelming, and maybe impossible. We don't know yet. If this becomes difficult, this is not the burden of devs working on the Upstage project. 

Please @heluDuyne @teddyfullstack , be aware of this. Feel free to find the simplest way to implement something FIRST, and get that working, saving complexity for later, or even **never.**

**Comment by vickis** [comment]
agree - take care of your capacity team :)

**Comment by helenvarley** [comment]
commenting here as gloria & i touched on this yesterday, when testing whether or not individual streams could be saved in scenes (they can, but only while you're on the stage; if you reload or come back later, the scene does not include your stream). we discussed how there would have to be an object for that to work, which is also what elisa has noted above about making a way for a stream to be sent from a mobile, without actually being on the stage: there needs to be an object for that stream that another player can add to the stage.

& as gloria has noted above, this could be an overwhelming, if not impossible, challenge ... but of the 3 things i listed at the top as being very useful to be able to do from a phone, streaming is definitely the most important one ...
